lib/operations/mongo_client_ops.js

```diff
@@ -581,12 +582,15 @@ function transformUrlOptions(_object) {
   }
 
   if (_object.auth) {
-    object.auth = Object.assign({}, _object.auth);
-    if (object.auth.db) {
-      object.authSource = object.authSource || object.auth.db;
-    }
-    if (object.auth.username) {
-      object.auth.user = object.auth.username;
+    if (_object.auth.username || (_object.options && _object.options.authMechanism)) {
```

Comment:
a) Want to double check: if `_object.options.authMechanism` is present, but `_object.auth` isn't is our goal to just ignore the auth mechanism and not create credentials?

b) If this is the logic we're going with, then it looks like we have a nested `if` that could be flattened:

```js
if (_object.auth && (_object.auth.username || (_object.options && _object.options.authMechanism)))
```

that said, the full flattening is kind of ugly, so I feel like we might want to do

```js
const hasUsername = _object.auth && _object.auth.username;
const hasMechanism = _object.auth && _object.options && _object.options.authMechanism
if (hasUsername || hasMechanism) {
```

Since this is not a hot codepath (literally run once per client), I'm fine making this a little more verbose so it is easier to read.

Reply:
:+1: